[Fleet][dev] 06-15-preview

zman-ms opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5e91ff2d-1f5c-4e96-87c2-390e7451da37)

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [x] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

zman-ms committed:
Adds base for updating Microsoft.ContainerService/fleet from version preview/2023-03-15-preview to version 2023-06-15-preview

zman-ms committed:
Updates readme

zman-ms committed:
Updates API version in new specs and examples

JainRah committed:
Double Header Fix (#24760)

brandom-msft committed:
[OpenAI.Inference] Adding content filter support (#24652)

* first-pass adding annotation

* FilterResult definitions

* move Filter definitions to common

* cleaning up naming and imports

* adding version headers

* added descriptions to objects

* correct self_harm JSON projection

* Correct to "prompt_annotations"

* "prompt_annotations" for chat, too

* Update specification/cognitiveservices/OpenAI.Inference/models/completions.common.tsp

Co-authored-by: Shawn Fang <45607042+mssfang@users.noreply.github.com>

* Add per-choice content_filter_results to /completions

* tsp compile after merge

* recompile with 0.46.0 after today's update

---------

Co-authored-by: Travis Wilson <35748617+trrwilson@users.noreply.github.com>
Co-authored-by: Shawn Fang <45607042+mssfang@users.noreply.github.com>

lmazuel committed:
Update OpenAI PR assignment (#24794)

* Update OpenAI PR assignment

* Update pull_request_assignment.yml

ckairen committed:
[typespec-ci] Automatically detect spec folders to validate (#24629)

weidongxu-microsoft committed:
[fist] mgmt, update service name (#24781)

mikeharder committed:
[typespec-pr] Add pipeline (#24802)

mikeharder committed:
[typespec-ci] Remove path wildcard (#24804)

weshaggard committed:
Update engsys CODEOWNERS (#24808)

trrwilson committed:
Align optionality of content filter categories (#24819)

Commit:
Add spec ARM PR review workflow diagram to ARM PR description template. (#24823)

* Add spec PR review workflow diagram to ARM PR description template.

Commit:
Update control_plane_template.md: fix link to diagram (#24828)

jpalvarezl committed:
Added generator config for OpenAI `Functions` for the Java SDK (#24765)

maximrytych-ms committed:
Remove repeatability support for token revocation endpoint (#24589)

dhu5432 committed:
Update clientTYpe (#24503)

KalpeshChavan12 committed:
[Hub Generated] Publish private branch 'dev-maintenance-Microsoft.Maintenance-2023-04-01' (#24048)

* Revert "Revert "[Hub Generated] Publish private branch 'dev-maintenance-Microsoft.Maintenance-2023-04-01' (#23832)" (#23923)"

This reverts commit e6c0e188a74d051a3a44728ce058e85447553392.

* Removed 202 and Async supression.

* Update documentation.

* Updated documentation

---------

Co-authored-by: Kalpesh Chavan <kachavan@microsoft.com>

mikeharder committed:
[package-lock.json] Add names to local dependencies (#24846)

- Command "npm update" removes these names, but "npm install" adds them
- It's probably better to keep the names, since users are more likely to run "npm install"

Commit:
Update ci-fix.md: add API Doc Preview TSG (#24843)

* Update ci-fix.md: add API Doc Preview TSG

kashyap-kunal committed:
Remove unsupported SQL Pool Rename/move operation (#24517)

ishanjindalms committed:
Updated text for property isSoftDeleteFeatureStateEditable (#24713)

athulvijayanmsft committed:
Updating folder structure for Microsoft.Migrate (#24550)

* folders restructure and update readme

* revert removed offazure in readme

thomasricci committed:
Removing readonly flag from top level object (#24848)

zman-ms committed:
sync with ADO repo

sarkar-rajarshi committed:
[ACS JobRouter] Add misssing descriptions (#24842)

* Updated JR swagger with maxNumberOfJobs

* sync with upstream main

* add descriptions

* prettier fix

* add more missing descriptions

* add more description fixes

---------

Co-authored-by: williamzhao87 <williamzhao87@gmail.com>

trrwilson committed:
a couple of representation tweaks for codegen (#24847)

serbrech commented:
<html>
<body>
<!--StartFragment-->

Rule | Message
-- | --
❌ 1011 - AddingResponseCode | The new version adds a response code '202'.New: fleet/preview/2022-09-02-preview/fleets.json#L358:11
❌ 1011 - AddingResponseCode | The new version adds a response code '202'.New: fleet/preview/2023-03-15-preview/fleets.json#L361:11
❌ 1011 - AddingResponseCode | The new version adds a response code '202'.New: fleet/preview/2023-03-15-preview/fleets.json#L785:11
❌ 1044 - XmsLongRunningOperationChanged | The new version has a different 'x-ms-longrunning-operation' value than the previous one.New: fleet/preview/2022-09-02-preview/fleets.json#L307:7Old: fleet/preview/2022-09-02-preview/fleets.json#L307:7
❌ 1044 - XmsLongRunningOperationChanged | The new version has a different 'x-ms-longrunning-operation' value than the previous one.New: fleet/preview/2023-03-15-preview/fleets.json#L310:7Old: fleet/preview/2023-03-15-preview/fleets.json#L310:7
❌ 1044 - XmsLongRunningOperationChanged | The new version has a different 'x-ms-longrunning-operation' value than the previous one.New: fleet/preview/2023-03-15-preview/fleets.json#L724:7Old: fleet/preview/2023-03-15-preview/fleets.json#L706:7

These breaking changes are **known**. We are adding an async patch implementation to the api. ManagedIdentity portal integration is built for patch async. we want to take this breaking change now in preview.
this will cause an SDK breaking change for the 06-15-preview sdks, but does not break at the API level for all existing versions.
the SDKs will not be re-generated.

Typespec does not provide a good way to version this change, so we'd like to apply it to all for the sake of future maintenance of the service.

jpalvarezl committed:
Removed FunctionDefinition as custom class (#24870)

msyyc committed:
Update readme.python.md (#24873)

* Update readme.python.md

* Update readme.python.md

kazrael2119 committed:
add title in typescript.md (#24879)

guoqing2023 committed:
Ericasp/number lookup preview (#24787)

* add isoCountryCode field (#24223)

* changed isoCountryCode description

---------

Co-authored-by: Erica <34174887+ericasp16@users.noreply.github.com>
Co-authored-by: Erica Sponsler <ericasp@microsoft.com>

LuChen-Microsoft committed:
Update communicationserviceschat.json (#24845)

kazrael2119 committed:
update client name in typescript.md (#24894)

kazrael2119 committed:
update sphere ts.md (#24895)

xiaoxuqi-ms committed:
Convert previous preview version Private Link to GA: 2023-06-30 for mysql (#24738)

* copy previous version as base for 2023-06-30

* update api version

* update readme

* fix arm id

MikhailTryakhov committed:
Fixed code owner for NRP (#24889)

serbrech commented on specification/containerservice/Fleet.Management/helpers.tsp:
```diff
@@ -24,96 +24,52 @@ alias IfNoneMatchParameters<T extends Azure.ResourceManager.Foundations.ArmResou
 @post
 op ArmResourceNoRequestBodyActionAsync<
   TResource extends Azure.ResourceManager.Foundations.ArmResource,
-  TResponse extends TypeSpec.Reflection.Model,
```
revert this line change

serbrech commented on specification/containerservice/Fleet.Management/package.json:
remove file from PR

serbrech commented on specification/containerservice/Fleet.Management/package-lock.json:
remove file from PR

serbrech commented on specification/containerservice/Fleet.Management/tspconfig.yaml:
revert this `git checkout specification/containerservice/Fleet.Management/tspconfig.yaml`

Commit:
Synapse SDK release [2023-06] (#24908)

anandanthony committed:
Anfranci/fixstartexectemplate2 (#24885)

* fix start job exec template required property

* fix start job exec template required property

* fix start job exec template required property

usharma6 committed:
SRP Swagger Updates for Jan23 [2023-01-01] Api Version GA  (#23918)

* Full copy from 2022-09-01 to 2023-01-01 without any changes

* update examples api version, update swagger api version for each swagger, update readme files

* Feature: AutoUptier for LCM with example

* prettier check

* Update new deafults for allowBlobPublicAccess and allowCrossTenantReplication

* Feature: ObjectReplicationRestorePoint APIs with examples, validation and prettier check

* Feature: Customer Initiated Migration with validation and prettier check

* Feature: Planned Failover with validation and prettier check

* Feature: BlobInventoryPolicyFilter creation time with validation and prettier check

* Revert "Feature: ObjectReplicationRestorePoint APIs with examples, validation and prettier check"

This reverts commit 6942ddfc948f9693cd04e4768a361a6ab0607332.

* Address comments for autoUptier

* address Migration comments

* Update allowedMethods for CORS

* Revert "Address comments for autoUptier"

This reverts commit 5ac6d025ef1d2c5b5898f9e102d6f1cfa4271840.

* Revert "Feature: AutoUptier for LCM with example"

This reverts commit a1469d0b51914d11058e0cab8a7cc294e5aae761.

* remove autoUptier example

* targetSkuName description and migration name parameter

* Remove Get customer initiated migration

* Fix targetSkuName required property, fix typo

* Address lint diff new errors

* Add get migration back, fix uri for post migration

* remove query parameter from post migration

* Resolve Lint Diff errors

* Spell Chek errors

* Fix Post Migration example to return header

---------

Co-authored-by: Udit Sharma <uditsharma@microsoft.com>

FumingZhang committed:
[Hub Generated] Review request for Microsoft.ContainerService/aks to add version stable/2023-06-01 (#24581)

* Adds base for updating Microsoft.ContainerService/aks from version stable/2023-05-01 to version 2023-06-01

* Updates readme

* Updates API version in new specs and examples

* Hobov2 Swagger (#24471)

* Hobov2 Swagger

* fix error

* update python & java readmes (#24462)

* Bring Node os upgrade channel up GA from preview except security patch (#24487)

* bring nodeosimage up to GA except for security channel

* Update managedClusters.json

* add vpa (#24508)

Co-authored-by: zhifanhui <zhifanhui@microsoft.com>

* Allow user to specify draintimeout in ga api (#24488)

---------

Co-authored-by: Xu Xue <55420084+xuexu6666@users.noreply.github.com>
Co-authored-by: Paul Miller <paul.miller@gmail.com>
Co-authored-by: huizhifan <60024380+huizhifan@users.noreply.github.com>
Co-authored-by: zhifanhui <zhifanhui@microsoft.com>
Co-authored-by: Paul Miller <pmiller@microsoft.com>

Joanna-Yang-Art committed:
Release network microsoft.network official 2023 04 01 2 (#24841)

* Adds base for updating Microsoft.Network from version stable/2023-02-01 to version 2023-04-01

* Updates readme

* Updates API version in new specs and examples

* Double Header Fix (#24783)

* Add Basic sku to applicationGateway.json (#24717)

* Add http listeners alias to frontend ip config (#24718)

* Sync mode property on Load balancer backend pool (#24685)

* Sync mode property on Load balancer backend pool

* model fix

---------

Co-authored-by: unknown <anjbal@DESKTOP-ML6T5PL.redmond.corp.microsoft.com>

* Migrate to IP based LB API change (#24633)

* Migrate to IP based LB API change

* Nit Fix

* NIT Fix

* Add body param pools

* Param name update

---------

Co-authored-by: unknown <anjbal@DESKTOP-ML6T5PL.redmond.corp.microsoft.com>

---------

Co-authored-by: Mikhail <mitryakh@microsoft.com>
Co-authored-by: JainRah <108508612+JainRah@users.noreply.github.com>
Co-authored-by: russsantos <127546568+russsantos@users.noreply.github.com>
Co-authored-by: anjbal <137822143+anjbal@users.noreply.github.com>
Co-authored-by: unknown <anjbal@DESKTOP-ML6T5PL.redmond.corp.microsoft.com>

RheaHooda23 committed:
Adding new APIs for billing visibility in new version 2023-07-01-preview for Microsoft.Elastic (#24587)

* Base API Commit

* Changing API version to 2023-07-01-preview

* Adding new apis for billing visibility

* updating readme.md

* added partnerEntityUri in the response

* model validation fixed

* taking parameters from v5 common types

ckairen committed:
[typespec-ci] Improve change detection (#24825)

- Fixes #24813
- Fixes #24818

ckairen committed:
[typespec - pr] Validate all specs if any changes to common files (#24826)

- Fixes #24814

mikeharder committed:
[typespec-ci] Disable PR trigger (#24920)

seanmcc-msft committed:
Storage STG 90 (#24915)

* Added new version of Data Lake swagger (#23905)

* Added new Files swagger (#23907)

* Paginated Directory Delete (#24146)

weidongxu-microsoft committed:
mgmt, java, azurestackhci, dedup name "Update" (#24924)

kazrael2119 committed:
update parameter name in selfhelp sample (#24928)

allenjzhang committed:
Update TypeSpec get started guide (#24901)

* Update TypeSpec get started guide

* Update documentation/typespec-rest-api-dev-process.md

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

---------

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

orenkaf committed:
[Hub Generated] Review request for Products to add version preview/2023-05-01-preview (#24678)

* Adds base for updating Products from version preview/2023-01-01-preview to version 2023-05-01-preview

* Updates readme

* Updates API version in new specs and examples

* 2023-05-01-preview version

* fix locations

* fix readme

* fix validations 1

* fix RAM

* Add has RI

* Remove dynamic page size query param

* Refactor operation id

* Fix readme

* Fix products operation id

* Documentation fix 1

* Documentation fix 2

* Documentation fix 3

* spell check fixes

* prettier fixes

---------

Co-authored-by: Oren Kafka <orkafka@microsoft.com>

ckairen committed:
[Get-TypeSpec-Folders.ps1] Sort output alphabetically (#24932)

ckairen committed:
[Validate-TypeSpec.ps1] Add GitClean switch (#24933)

- Continuation of #24629

mbashtovaya committed:
[Hub Generated] Publish private branch 'RPSaaSMaster' - NetworkCloud 2023-07-01 stable API  (#24722)

* add or modify files

* removed non-published old private preview

carlowisse committed:
New stable api 2023-04-01 Microsoft.RedHatOpenShift (#23772)

* Previous API version

* New API 2023-04-01

* Fix swagger linting error

* Added xmsEnum for OutboundType

* Re-word description for OutboundType

dulikvor committed:
Microsoft.OperationalInsights: Correcting Total retention in days limitation. (#24574)

* OperationalInsights: 2022-10-01 stable version

* Actual changes.

* More fixes.

* more changes.

* Removing redundant breaking change.

* Workspace identity: referencing standard types.

* Referencing managed identity standard type, refactoring booleans into enum.

* Fixing examples.

* Removing duplicate resource.

* Update Workspaces.json

* Update Clusters.json

* Revert "Update Clusters.json"

This reverts commit 4eb99cf4699afeb95dc2bab1ee071c7d4f9ee2c5.

* Revert "Update Workspaces.json"

This reverts commit c16bc82f2ec8a35131c1ec54d2dca147eca929d1.

* Returning to our identity implementation.

* OperationalInsights 2022-10-01: Changing retentionInDaysAsDefault to boolean representation.

* redundant comma.

* redundant comma.

* Adding missing example header.

* Update TablesDelete.json

* Microsoft.OperationalInsights: Correcting Total retention in days limitation.

Alancere committed:
config reademe.go.md (#24941)

* config reademe.go.md

* format

kuanlu95 committed:
[Hub Generated] Publish private branch 'main' (#23608)

* add swagger

* updated swagger

* Updated readme for Unified Vision Service

* fixed build error

* update api version

* fixed typo

* update comment

* Add examples

* fixed

* prettier

* add custom word

* Address some comments

---------

Co-authored-by: KUAN LU <kualu@microsoft.com>

zman-ms committed:
exclude unneeded file changes

zman-ms committed:
revert tsp config

zman-ms committed:
revert helpers.tsp

zman-ms committed:
remove package files

zman-ms committed:
remove package files

zman-ms committed:
add known word containerd

serbrech commented on specification/containerservice/Fleet.Management/.npmrc:
remove this file

serbrech commented on package-lock.json:
revert this

serbrech commented on specification/containerservice/Fleet.Management/main.tsp:
```diff
@@ -19,12 +19,15 @@ namespace Microsoft.ContainerService;
   @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
   v2023_03_15_preview: "2023-03-15-preview",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
```
are there newer versions of these dependencies?

serbrech commented on specification/containerservice/Fleet.Management/updaterun.tsp:
```diff
@@ -21,50 +21,49 @@ namespace Microsoft.ContainerService;
+    @visibility("create", "read")
+    name: string;
+    ...EntityTag;
 }
```
why is the indentation changing?

zman-ms replied:
Ran tsp format and fixed.

MattEberhart-MSFTE committed:
Changing operation name to match other related API for documentation (#24588)

navba-MSFT committed:
Update computeRPCommon.json (#24910)

zman-ms committed:
format

Alancere committed:
add directive mobilenetwork reademe.go.md (#24942)

zman-ms committed:
remove .npmrc

zman-ms committed:
revert custom word

zman-ms committed:
revert package-lock.json

Alancere committed:
sql readme.go.md directive (#24610)

* sql readme.go.md directive

* rename to ServerPrivateEndpointConnectionProperties

* lenient-model-deduplication

saifahmed098 committed:
Saifahmed098 databoxedge microsoft.data box edge 2023 07 01 (#24926)

* Adds base for updating Microsoft.DataBoxEdge from version stable/2022-03-01 to version 2023-07-01

* Updates readme

* Updates API version in new specs and examples

* Adding string KubernetesWorkloadProfile in databoxedgeProperties

zman-ms committed:
re-generate json

zman-ms committed:
add example

zman-ms committed:
add example

zman-ms committed:
fix read-only fields

zman-ms committed:
fix read-only fields

zman-ms committed:
re-generate json with latest compiler

zman-ms committed:
add custom word back

zman-ms committed:
re-generate json

zman-ms committed:
restore config

zman-ms committed:
add examples

zman-ms committed:
add examples

zman-ms committed:
add examples

zman-ms committed:
add examples

zman-ms committed:
add examples

zman-ms committed:
add examples

zman-ms committed:
prettier

zman-ms committed:
fix examples

zman-ms committed:
git ignore

serbrech commented on specification/containerservice/Fleet.Management/.gitignore:
```diff
@@ -0,0 +1,5 @@
+node_modules
+output
+tspconfig.yaml
```
Shouldn't ignore tspconfig.yaml, but should have the same as what's currently in the main branch.

serbrech commented on specification/containerservice/Fleet.Management/examples/2022-09-02-preview/Fleets_PatchTags.json:
```diff
@@ -46,6 +46,12 @@
+    "202": {
+      "headers": {
+        "Retry-After": "20",
+        "Location": "East US"
```
location is not East US, it's the operationStatus url. 
See Delete example :

```
  "Location": "https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ContainerService/locations/eastus/operations/00000000-0000-0000-0000-000000000000?api-version=2022-02-01"
  ```

zman-ms committed:
review commoents

zman-ms committed:
fix examples

msyyc committed:
Update readme.python.md (#24980)

* Update readme.python.md

* Update readme.python.md

fhaghbin-msft committed:
[ACS - Call Automation] [Beta2] Api spec for 2023-06-15-preview  (#24482)

* Add beta2 api spec

* add readme

* fix lines

* Fix readme

* Added repeatability

* Fix 202 long running

* Fix header

* Fix the base

* Fix check

* added x-ms-long-running-operation

* add location

* fix example

* test

* test

* test

* fix

* test

* fix test

* test

* Fix

* test

* test

* fix

* Remove LRO

* Updates as per ARB feedback

* Updates as per ARB feedback

* Fix senddtmfTones

* Fix error

* Fix send dtmf description

Commit:
Update/remove obsolete docs (#24946)

zman-ms committed:
make node image selection optional

bexxx committed:
Add openapi docs for OpenAI api version 2023-07-01-preview (#24992)

zman-ms committed:
fix warnings

zman-ms committed:
fix warnings

zman-ms committed:
fix warnings

zman-ms committed:
fix docs

XiaofeiCao committed:
rename EdgeZoneStorageAccountType back to StorageAccountType to avoid breaking (#24988)

yareyes committed:
Adding new optional parameter to Auto Patching (#24554)

mikeharder committed:
[TypeSpec Validation] Get-TypeSpec-Folders.ps1 should only validate all specs on selected files (#25010)

- Fixes #25006

msyyc committed:
Update readme.python.md (#25009)

* Update readme.python.md

* Update readme.python.md

orenkaf committed:
[Hub Generated] Review request for Products to add version preview/2023-05-01-preview (#25014)

* Documentation fixes

* Documentation fixes 2

* Documentation fixes 3

* Documentation fixes 4

* Documentation fixes 5

---------

Co-authored-by: Oren Kafka <orkafka@microsoft.com>

jorgecotillo committed:
fix enum mismatch (#24559)

thramu committed:
Moving Markup rules API into separate file (#25003)

angiurgiu committed:
[ACS Chat] [Rest API] Added chat thread metadata and participant meta… (#24837)

* [ACS Chat] [Rest API] Added chat thread metadata and participant metadata. (#24243)

* Updating ACS Chat swagger with Chat thread and participant metadata.

* Fixed swagger version.

* Fixed swagger version.

* Re-added data retention policy to the swagger.

* Update communicationserviceschat.json - changed metadata from string to dictionary

* Update communicationserviceschat.json

* Update communicationserviceschat.json

Corrected size limitations - should be 1k, not 2k

* Update communicationserviceschat.json

Added metadata to patch thread request

* Update communicationserviceschat.json

---------

Co-authored-by: Andrei Giurgiu <agiurg@microsoft.com>

* remove data retention from 2023-07-01

---------

Co-authored-by: Andrei Giurgiu <agiurg@microsoft.com>
Co-authored-by: Lu Chen <luc@microsoft.com>

zman-ms committed:
Adds base for updating Microsoft.ContainerService/fleet from version preview/2023-03-15-preview to version 2023-06-15-preview

zman-ms committed:
Updates readme

zman-ms committed:
Updates API version in new specs and examples

zman-ms committed:
revert package changes

zman-ms committed:
Merge branch 'dev-aks-fleet-2023-06-15-preview' into zman/fleet-api-0615-changes

zman-ms committed:
compile
